tests/core/test_validation_config.py

```diff
@@ -0,0 +1,358 @@
+from __future__ import annotations
+
+import json
+import uuid
+from unittest import mock
+
+import pytest
+
+from great_expectations.core.batch_config import BatchConfig
+from great_expectations.core.expectation_suite import ExpectationSuite
+from great_expectations.core.validation_config import ValidationConfig
+from great_expectations.data_context.data_context.ephemeral_data_context import (
+    EphemeralDataContext,
+)
+from great_expectations.datasource.fluent.pandas_datasource import (
+    CSVAsset,
+    PandasDatasource,
+)
+
+
+class TestValidationConfigSerialization:
+    ds_name = "my_ds"
+    asset_name = "my_asset"
+    batch_config_name = "my_batch_config"
+    suite_name = "my_suite"
+    validation_config_name = "my_validation"
+
+    @pytest.fixture
+    def validation_config_data(
+        self,
+        in_memory_runtime_context: EphemeralDataContext,
+    ) -> tuple[PandasDatasource, CSVAsset, BatchConfig]:
+        context = in_memory_runtime_context
+
+        ds = context.sources.add_pandas(self.ds_name)
+        asset = ds.add_csv_asset(self.asset_name, "data.csv")
+        batch_config = asset.add_batch_config(self.batch_config_name)
+
+        return ds, asset, batch_config
+
+    @pytest.fixture
+    def validation_config_suite(self) -> ExpectationSuite:
+        return ExpectationSuite(self.suite_name)
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize(
+        "ds_id, asset_id, batch_config_id",
+        [
+            (
+                "9a88975e-6426-481e-8248-7ce90fad51c4",
+                "9b35aa4d-7f01-420d-9d45-b45658e60afd",
+                "782c4aaf-8d56-4d8f-9982-49821f4c86c2",
+            ),
+            (None, None, None),
+        ],
+        ids=["with_data_ids", "without_data_ids"],
+    )
+    @pytest.mark.parametrize(
+        "suite_id",
+        ["9b35aa4d-7f01-420d-9d45-b45658e60afd", None],
+        ids=["with_suite_id", "without_suite_id"],
+    )
+    @pytest.mark.parametrize(
+        "validation_id",
+        ["708bd8b9-1ae4-43e6-8dfc-42ec320aa3db", None],
+        ids=["with_validation_id", "without_validation_id"],
+    )
+    def test_validation_config_serialization(
```

Comment:
For all of these tests, mocking felt like quite a burden (especially with JSON serialization issues). I opted for using real objects and relationships but if there are improvements to be made here to better isolate the unit under test, let's talk 😄